### PR TITLE
UMM-G support (JW)

### PR DESCRIPTION
**Pull request by jenny-m-wood**
Added rule_mappings for umm-g fields 

- Characteristic Value Check
- MIME type Check
- Instrument Short Name GCMD Check
- Platform Short Name GCMD Check
- Data Format GCMD check
- Campaign Short Name GCMD check
- Campaign Name Presence Check
- Online Resource URL Description Check
- Spatial Extent Requirement Fulfillment Check
- Online Resource Type GCMD Check
- Characteristic Name Uniqueness Check
- GET DATA URL Check

### CODE REVIEW

File: pyQuARC/schemas/rule_mapping.json [hunk 1]
```diff
@@ -3631,7 +3730,7 @@
     "get_data_url_check_umm": {
         "rule_name": "GET DATA URL check for UMM JSON",
         "fields_to_apply": {
-            "umm-json": [
+            "umm-g": [
```
Comment (slesaad) [inline pyQuARC/schemas/rule_mapping.json:3733]: why'd you replace this? "umm-json" holds the ummc checks. by doing this, you are removing the ummc check and adding a ummg check. you probably want both checks, right? @jenny-m-wood

Comment (jenny-m-wood) [reply pyQuARC/schemas/rule_mapping.json:3733]: @slesaad yep, that's a mistake on my part. Thanks for catching it!

**Review by svbagwell — APPROVED** [review]

**Comment by spa0002** [comment]
In rule_mapping.json -
For `characteristic_name_uniqueness_check`: for umm-g, see "Characteristics" as property of "InstrumentType" but  not "PlatformType" in umm-g-json-schema.json

**Review by spa0002 — COMMENTED** [review]
In rule_mapping.json -
For characteristic_name_uniqueness_check: for umm-g, see "Characteristics" as property of "InstrumentType" but not "PlatformType" in umm-g-json-schema.json

**Review by spa0002 — APPROVED** [review]

**Review by esr0004 — COMMENTED** [review]
1. characteristic_value_check: 
The umm-g-json-schema.json has "Characteristics" as property of "InstrumentType" but not "PlatformType" - Recommend removing "Platforms/Characteristics" fields.

2. data_format_gcmd_check:
Recommend adding “DataGranule/ArchiveAndDistributionInformation/Files/Format” to rule_mapping.json

3. spatial_extent_requirement_fulfillment_check:
Recommend capitalizing the “p” in “Gpolygons” for the path inside the rule_mapping.json so it reads “SpatialExtent/HorizontalSpatialDomain/Geometry/GPolygons" (also present in UMM-C field)

4. characteristic_name_uniqueness_check_umm:
Recommend adding "Platforms/Instruments/ComposedOf/Characteristics" to rule_mapping.json

**Review by esr0004 — APPROVED** [review]

File: pyQuARC/code/custom_validator.py [hunk 2]
```diff
@@ -15,15 +15,15 @@ def ends_at_present_flag_logic_check(
         ends_at_present_flag, ending_date_time, collection_state
     ):
         collection_state = collection_state.upper()
-        if ends_at_present_flag == None:
+        if not ends_at_present_flag:
```
Comment (slesaad) [inline pyQuARC/code/custom_validator.py:18]: this is incorrect. this will execute if the `ends_at_present_flag` is `False` too, which isn't what we want

**Pull request merged**